### PR TITLE
[INLONG-6747][Sort] StarRocks connector supports transferring all tables for all schemas in one database 

### LINKED ISSUE #6747: [Feature][Sort]StarRocks connector supports transferring all tables for all schemas in one database
### Description

Inspired by doris-cdc-inlong, starrocks-cdc-inlong can also support transferring all tables in one database on time.

StarRocks FE api has changed, been different from Doris FE api. If you use doris-cdc-inlong to handle StarRocks tables writing, it will give a error: http://x.x.x.x:xxxx/api/backends?is_alive=true failed. java.io.IOException: Failed to get response from Doris.

That is why we need a sarrocks-cdc-inlong connector.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaorui**
### Prepare a Pull Request

- [INLONG-6747][Sort] StarRocks connector supports transferring all tables for all schemas in one database 

- Fixes #6747

### Motivation

Inspired by doris-inlong, starrocks-inlong can also support transferring all tables in one database on time.

StarRocks FE api has changed, is different from Doris FE api. If you use doris-cdc-inlong to handle StarRocks tables writing, it will give a error: `http://x.x.x.x:xxxx/api/backends?is_alive=true failed. java.io.IOException: Failed to get response from Doris`. That is why we need a starrocks-inlong connector.

Based on flink-starrocks from community, starrocks-inlong supports all tables writing in one database one time.

### Modifications

1. StarRocksDynamicSinkFunction's invoke method supports handling debezium canal-json upstream parsing. It picks data from upstream, assembles data with json format.

2. StarRocksSinkManager's init method will only validate table structure in single table transferring, and skip validating in all database migration.

3. StarRocksStreamLoadVisitor's checkLabelState method add String database parameter.

4. StarRocksSinkManager removes checkFlushException method invoking in waitAsyncFlushingDone, close and flushInternal methods. If there is a error when writing to StarRocks, flink cdc will fail to make a checkpoint because of this error.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/manager/StarRocksStreamLoadVisitor.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.DefaultRedirectStrategy;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StarRocksStreamLoadVisitor implements Serializable {
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/manager/StarRocksStreamLoadVisitor.java:53]: Please add license

**Comment by gong** [comment]
@liaorui Please add packaging config for this connector in `inlong-distribution/src/main/assemblies/sort-connectors.xml`

**Comment by gong** [comment]
@liaorui Please add base shade in starrocks.pom, ref to iceberg pom. Because It can avoid modifying base lead classloader problem.
```
     <relocations>
                                <relocation>
                                    <pattern>org.apache.inlong.sort.base</pattern>
                                    <shadedPattern>
                                        org.apache.inlong.sort.starraocks.shaded.org.apache.inlong.sort.base
                                    </shadedPattern>
                                </relocation>
                            </relocations>
```

**Comment by liaorui** [comment]
> @liaorui Please add packaging config for this connector in `inlong-distribution/src/main/assemblies/sort-connectors.xml`

done

**Comment by liaorui** [comment]
> @liaorui Please add base shade in starrocks.pom, ref to iceberg pom. Because It can avoid modifying base lead classloader problem.
> 
> ```
>      <relocations>
>                                 <relocation>
>                                     <pattern>org.apache.inlong.sort.base</pattern>
>                                     <shadedPattern>
>                                         org.apache.inlong.sort.starraocks.shaded.org.apache.inlong.sort.base
>                                     </shadedPattern>
>                                 </relocation>
>                             </relocations>
> ```

done

**Review by gong — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**